### PR TITLE
First pass at using RTM client

### PR DESCRIPTION
**Pull request by TRManderson**
Again this is untested code. Hopefully works fine. It at least removes a few deps.

### CODE REVIEW

File: uqcsbot/__init__.py [hunk 1]
```diff
@@ -19,9 +20,11 @@ def main():
 
     # Run bot
     if '--dev' in sys.argv:
+        logging.basicConfig(level=logging.DEBUG)
```
Comment (CameronAavik) [inline uqcsbot/__init__.py]: I would rather the logging level toggle be a separate command, I do all the debugging directly against the slack API and would like the debug level in that circumstance

File: uqcsbot/base.py [hunk 2]
```diff
@@ -74,12 +83,25 @@ def _handle_command(self, event_data: dict) -> None:
 
     def on_command(self, command_name: str):
         def decorator(fn):
-            if not asyncio.iscoroutinefunction(fn):
```
Comment (CameronAavik) [inline uqcsbot/base.py]: Is on_command being deprecated? I see this code got moved to `register_handler` instead.

File: uqcsbot/base.py [hunk 3]
```diff
@@ -41,28 +39,39 @@ def async_worker(loop):
...
 
-    def _handle_command(self, event_data: dict) -> None:
-        message = event_data["event"]
+        self._handlers[''].append(self._handle_command)
+        self._handlers['hello'].append(self._handle_hello)
+        self._handlers['hello'].append(self._handle_goodbye)
+
+    def _handle_hello(self, evt):
+        if evt != {"type": "hello"}:
+            self.logger.debug(f"Goodbye event has unexpected extras: {evt}")
```
Comment (CameronAavik) [inline uqcsbot/base.py]: log should say "Hello" instead of "Goodbye"

File: uqcsbot/base.py (hunk 2 above)
Comment (TRManderson) [reply uqcsbot/base.py]: `on_command` is specific to the `Command` wrapper, and I think having such a wrapper is a really good idea. `on` (which uses `register_handler`), however, can be used to subscribe to any events whatsoever. So no, not deprecated, this is just a more specific API

File: uqcsbot/__init__.py (hunk 1 above)
Comment (TRManderson) [reply uqcsbot/__init__.py]: You OK with this being provisional and moving it out once we do nicer arg parsing? (perhaps with `click`). If so, I'll add a `# TODO`, if not I'll change it now.

Comment (CameronAavik) [reply uqcsbot/__init__.py]: Am fine with fixing it later.

File: uqcsbot/base.py (hunk 2 above)
Comment (CameronAavik) [reply uqcsbot/base.py]: How do non-async functions get called now? Since you are no longer wrapping them with run_async.

Comment (TRManderson) [reply uqcsbot/base.py]: Rip, was intending to replace this differently but evidently I didn't. Code review catches bugs yet again!

Comment (TRManderson) [reply uqcsbot/base.py]: Fixed

**Comment by TRManderson** [comment]
Actually tested now!

**Pull request merged**